extension/app/ts/background/settings.ts

```diff
@@ -159,3 +161,26 @@ export async function savePendingInterceptorAccessRequestPromise(promise: Pendin
 	return await browser.storage.local.set({ InterceptorAccessRequestPromise: PendingInterceptorAccessRequestPromise.serialize(promise) })
 }
 
+export async function getSimulationResults() {
+	const results = await browser.storage.local.get(['simulationResults'])
+	if (results.simulationResults === undefined) {
+		return {
+			simulationId: 0,
+			simulationState: undefined,
+			visualizerResults: undefined,
+			addressBookEntries: [],
+			tokenPrices: [],
+			activeAddress: undefined
+		}
+	}
+	return SimulationResults.parse(results.simulationResults)
+}
+
+const simulationResultsSemaphore = new Semaphore(1)
+export async function updateSimulationResults(simulationResults: SimulationResults) {
+	simulationResultsSemaphore.execute(async () => {
+		const results = await browser.storage.local.get(['simulationResults'])
+		if (results.simulationResults !== undefined && simulationResults.simulationId < results.simulationResults.simulationId) return // do not update state with older state
+		return await browser.storage.local.set({ simulationResults: SimulationResults.serialize(simulationResults) })
+	})
+}
```

Comment:
Recommend using a constant for this type of thing so you don't have to worry about typos.

Also don't use `any`, and when a library uses it immediately cast to `unknown` to protect yourself from the terrible library.
```suggestion
const SIMULATION_RESULTS_STORAGE_KEY = 'simulationResults'
export async function getSimulationResults() {
	const results = await browser.storage.local.get([SIMULATION_RESULTS_STORAGE_KEY]) as unknown
	if (results.simulationResults === undefined) {
		return {
			simulationId: 0,
			simulationState: undefined,
			visualizerResults: undefined,
			addressBookEntries: [],
			tokenPrices: [],
			activeAddress: undefined
		}
	}
	return SimulationResults.parse(results.simulationResults)
}

const simulationResultsSemaphore = new Semaphore(1)
export async function updateSimulationResults(simulationResults: SimulationResults) {
	simulationResultsSemaphore.execute(async () => {
		const results = await browser.storage.local.get([SIMULATION_RESULTS_STORAGE_KEY]) as unknown
		// TODO: use funtypes to parse/validate `results` here.  `browser.storage.local.get` returns an `any` I think (which is terrible)
		if (results.simulationResults !== undefined && simulationResults.simulationId < results.simulationResults.simulationId) return // do not update state with older state
		return await browser.storage.local.set({ [SIMULATION_RESULTS_STORAGE_KEY]: SimulationResults.serialize(simulationResults) })
	})
}
```